packages/interactivity-router/src/head.js

```diff
@@ -0,0 +1,77 @@
+// Function to update only the necessary tags in the head.
```

Comment:
In the follow up work - a dedicated PR - it would be great to use JSDoc comments for all functions:

https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/coding-guidelines.md#javascript-documentation-using-jsdoc

Reply:
I added som JSDoc comments to the new functions introduced by this PR in [this commit](https://github.com/WordPress/gutenberg/pull/59707/commits/a2c1a070da65542dcf10d45e728c4be40010c56a). However, there are still missing docs in other interactivity API functions.